contentcuration/contentcuration/tasks.py

```diff
@@ -75,3 +123,45 @@ def generateusercsv_task(email):
 @task(name='deletetree_task')
 def deletetree_task(tree_id):
     ContentNode.objects.filter(tree_id=tree_id).delete()
+
+
+type_mapping = {
+    'test': {'task': test_task, 'progress_tracking': False},
+    'error-test': {'task': error_test_task, 'progress_tracking': False},
+    'progress-test': {'task': progress_test_task, 'progress_tracking': True},
+}
+
+
+def create_async_task(task_name, task_options, task_args=None):
```

Comment:
Can you add a docstring for this function? It looks important.

Reply:
Will do!

Reply:
Done!